### PR TITLE
docs(firebase_core): update docs about native SDK warnings

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

See title.

## Related Issues

fixes https://github.com/FirebaseExtended/flutterfire/issues/8119

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_core/firebase_core_platform_interface/lib/src/method_channel/method_channel_firebase.dart [hunk 1]
```diff
@@ -91,6 +91,8 @@ class MethodChannelFirebase extends FirebasePlatform {
           },
         ))!);
         defaultApp = appInstances[defaultFirebaseAppName];
+        // ignore: avoid_print
+        print('Initialized app from Dart application code.');
```
Comment (Salakar) [inline packages/firebase_core/firebase_core_platform_interface/lib/src/method_channel/method_channel_firebase.dart:95]: Lets remove this for now as discussed

File: docs/cli.mdx [hunk 2]
```diff
@@ -46,6 +46,12 @@ Additionally, if you use the [`google_sign_in` provider](./auth/social#google) f
 
 ### Generating Firebase Options
 
+:::note
+If you initialize your application from Dart, you will likely receive warnings from the native SDKs indicating no app has been configured or
```
Comment (Salakar) [inline docs/cli.mdx:50]:
```suggestion
If you initialize your application from Dart, you will likely receive warnings from the native Firebase SDKs indicating no Firebase app has been configured or
```

File: docs/cli.mdx [hunk 3]
```diff
@@ -46,6 +46,12 @@ Additionally, if you use the [`google_sign_in` provider](./auth/social#google) f
 
 ### Generating Firebase Options
 
+:::note
+If you initialize your application from Dart, you will likely receive warnings from the native Firebase SDKs indicating no Firebase app has been configured or
```
Comment (Salakar) [inline docs/cli.mdx:50]:
```suggestion
If you initialize your Firebase application from Dart, you will likely receive warnings from the native Android & iOS Firebase SDKs indicating no Firebase app has been configured or
```

**Comment by russellwheatley** [comment]
I've made a request for console warning update on iOS repo as well: https://github.com/firebase/firebase-ios-sdk/issues/9390

**Review by Salakar — APPROVED** [review]

**Pull request merged**